### PR TITLE
Integrate latest commits

### PR DESCRIPTION
**Pull request by jovial**

### CODE REVIEW

**Review by markgoddard — COMMENTED** [review]
I do wonder if having rally as a submodule might be a bit overly restrictive?

File: functions [hunk 1]
```diff
@@ -455,6 +455,18 @@ function workaround_start_sshd {
     fi
 }
 
+function workaround_git_config {
+    # Workaround: fatal: unable to auto-detect email address (got 'stack@runner-bkqvjp4m-project-21118777-concurrent-0.(none)')
+    # Might not be the best place to put this, but it is causing control_host_bootstrap to fail.
+    git config --global user.name "Kayobe Automation"
+    git config --global user.email "kayobe-automation@example.org"
```
Comment (markgoddard) [inline functions:462]: Does this require --global?

**Comment by jovial** [comment]
> I do wonder if having rally as a submodule might be a bit overly restrictive?

I think this still allows the use of custom images as you'd just customise the job that builds the container, e.g:

https://github.com/stackhpc/kayobe-automation/blob/main/examples/gitlab/.gitlab-ci.yml#L61

so I guess this serves as the reference image. Do you see any downsides with that approach?

**Comment by markgoddard** [comment]
> Edit: thanks for the review
> 
> > I do wonder if having rally as a submodule might be a bit overly restrictive?
> 
> I think this still allows the use of custom images as you'd just customise the job that builds the container, e.g:
> 
> https://github.com/stackhpc/kayobe-automation/blob/main/examples/gitlab/.gitlab-ci.yml#L61
> 
> so I guess this serves as the reference image. Do you see any downsides with that approach?

Why do we need rally as a submodule if it is used as a container?

File: functions (hunk 1 above)
Comment (jovial) [reply functions:462]: We could possibly get away with setting this for every git repository we clone, but I thought it represented a sensible default. I didn't actually expect a pip install to need this to be set and I didn't want to expose this complication to the user. We do override this to a user controlled value for any commits we actually make:

https://github.com/stackhpc/kayobe-automation/blob/main/functions#L269

**Comment by jovial** [comment]
> > Edit: thanks for the review
> > > I do wonder if having rally as a submodule might be a bit overly restrictive?
> > 
> > 
> > I think this still allows the use of custom images as you'd just customise the job that builds the container, e.g:
> > https://github.com/stackhpc/kayobe-automation/blob/main/examples/gitlab/.gitlab-ci.yml#L61
> > so I guess this serves as the reference image. Do you see any downsides with that approach?
> 
> Why do we need rally as a submodule if it is used as a container?

The submodule links to a repository that is used to build the image in which we run rally, it doesn't link to rally itself.

**Comment by markgoddard** [comment]
> > > Edit: thanks for the review
> > > > I do wonder if having rally as a submodule might be a bit overly restrictive?
> > > 
> > > 
> > > I think this still allows the use of custom images as you'd just customise the job that builds the container, e.g:
> > > https://github.com/stackhpc/kayobe-automation/blob/main/examples/gitlab/.gitlab-ci.yml#L61
> > > so I guess this serves as the reference image. Do you see any downsides with that approach?
> > 
> > 
> > Why do we need rally as a submodule if it is used as a container?
> 
> The submodule links to a repository that is used to build the image in which we run rally, it doesn't link to rally itself.

Oops, missed that.

File: functions (hunk 1 above)
Comment (markgoddard) [reply functions:462]: I think I'd be a bit surprised if I ran this locally then found I was suddenly making commits as someone called "Kayobe Automation" :)

Given that it's all scripted, if it's possible to set the config on a per-repo basis, I'd say that would be less invasive.

**Pull request merged**